samza-core/src/main/java/org/apache/samza/execution/JobPlanner.java

```diff
@@ -169,10 +170,13 @@ private JobConfig prepareTaskJob(TaskApplicationDescriptorImpl taskAppDesc) {
   }
 
   private Map<String, String> expandSystemStreamConfigs(ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc) {
+    // merge user-provided configuration with input/output descriptor generated configuration
+    // descriptor generated configuration has higher priority
     Map<String, String> systemStreamConfigs = new HashMap<>();
-    appDesc.getInputDescriptors().forEach((key, value) -> systemStreamConfigs.putAll(value.toConfig()));
-    appDesc.getOutputDescriptors().forEach((key, value) -> systemStreamConfigs.putAll(value.toConfig()));
-    appDesc.getSystemDescriptors().forEach(sd -> systemStreamConfigs.putAll(sd.toConfig()));
+    systemStreamConfigs.put(TaskConfigJava.BROADCAST_INPUT_STREAMS, config.get(TaskConfigJava.BROADCAST_INPUT_STREAMS));
```

Comment:
One more question: I have exclude the task.inputs from high-level applications (and for low-level TaskApplication), only exception is the LegacyTaskApplication. Shouldn't we do the same for task.broadcast.inputs when we allow InputDescriptor to specify broadcast option?